ide/src/decompose.ts

```diff
@@ -45,59 +46,53 @@ export let decompose = async (call_flowistry: CallFlowistry) => {
   let doc = active_editor.document;
   let selection = active_editor.selection;
 
-  try {
-    let cmd = `decompose ${doc.fileName} ${doc.offsetAt(selection.anchor)}`;
-    let decomp = await call_flowistry<Decomposition>(cmd);
-    if (!decomp) {
-      return;
-    }
+  let cmd = `decompose ${doc.fileName} ${doc.offsetAt(selection.anchor)}`;
+  let decomp = await call_flowistry<Decomposition>(cmd);
+  if (is_flowistry_error(decomp)) {
+    return decomp.show();
+  }
```

Comment:
This is applicable for all `call_flowistry` calls, but the idea is we can remove the `try { ... }` block by returning `T | FlowistryError`. The error is caught and displayed depending on its implementation of `show`.

My only worry about this is catching errors unrelated to flowistry (maybe raised by the extension?). It is also a matter of style preference so if you have thoughts I'd be very open.